src/test/scene.test.cpp

```diff
@@ -0,0 +1,212 @@
+}
+
+int test_remove() {
+
+    scene::Scene *scene = new scene::Scene();
+
+    vec2 pos1 = {0, 0};
+    CellObject cell1 = CellObject(pos1, 2, VK2D_BLUE);
+    vec2 pos2 = {50, 50};
+    CellObject cell2 = CellObject(pos2, 3, VK2D_RED);
+    vec2 pos3 = {100, 100};
+    CellObject cell3 = CellObject(pos3, 1, VK2D_BLACK);
+
+    auto go1 = scene::GameObject::createFromDerived(cell1);
+    auto go1_1 = scene::GameObject::createFromDerived(cell1);
+    auto go2 = scene::GameObject::createFromDerived(cell2);
+    auto go3 = scene::GameObject::createFromDerived(cell3);
+    scene->setDupsAllowed(true);
+    scene->addObject(go1);
+    scene->addObject(go1);
+    scene->addObject(go1);
+    scene->addObject(go1);
+    scene->addObject(go1);
+    scene->addObject(go2);
+    scene->addObject(go3);
+
+    if (scene->sizeObjects() != 7) {
+        return EXIT_FAILURE;
+    }
+
+    scene->removeOneObject(go1);
+    if (scene->sizeObjects() != 6) {
+        return EXIT_FAILURE;
+    }
+
+    scene->removeAllObject(go1);
+    if (scene->sizeObjects() != 2) {
+        return EXIT_FAILURE;
+    }
+
+    scene->addObject(go1);
+    scene->addObject(go1);
+    scene->addObject(go1_1);
+    scene->addObject(go1_1);
+    if (scene->sizeObjects() != 6) {
+        return EXIT_FAILURE;
+    }
+
+    scene->removeAllObject(go1);
+    if (scene->sizeObjects() != 2) {
+        return EXIT_FAILURE;
+    }
+
+    return EXIT_SUCCESS;
+}
+
+int test_exist() {
+
+    scene::Scene *scene = new scene::Scene();
+
+    vec2 pos1 = {0, 0};
+    CellObject cell1 = CellObject(pos1, 2, VK2D_BLUE);
+    vec2 pos2 = {50, 50};
+    CellObject cell2 = CellObject(pos2, 3, VK2D_RED);
+
+    auto go1 = scene::GameObject::createFromDerived(cell1);
+    auto go1_1 = scene::GameObject::createFromDerived(cell1);
+    auto go2 = scene::GameObject::createFromDerived(cell2);
+    scene->addObject(go1);
+    scene->setDupsAllowed(false);
+
+    bool exist = scene->existObject(go1);
+    if (exist != true) {
+        return EXIT_FAILURE;
+    }
+    exist = scene->existObject(go1_1);
+    if (exist != true) {
+        return EXIT_FAILURE;
+    }
+
+    exist = scene->existObject(go2);
+    if (exist != false) {
+        return EXIT_FAILURE;
+    }
+
+    scene->removeOneObject(go1);
+    exist = scene->existObject(go1);
+    if (exist != false) {
+        return EXIT_FAILURE;
+    }
+
+    return EXIT_SUCCESS;
+}
+
+int main(int argc, char *argv[]) {
+
+    int res = test_add();
+    if (res) {
+        printf("test_add FAILURE\n");
+        return EXIT_FAILURE;
+    }
+    printf("test_add SUCCESS\n");
+
+    res = test_remove();
+    if (res) {
+        printf("test_remove FAILURE\n");
+        return EXIT_FAILURE;
+    }
+    printf("test_remove SUCCESS\n");
+
+    res = test_exist();
+    if (res) {
+        printf("test_exist FAILURE\n");
+        return EXIT_FAILURE;
+    }
+    printf("test_exist SUCCESS\n");
+
+    printf("all complete\n");
+    return EXIT_SUCCESS;
```

Comment:
Instead of exiting on a failed test, which would mean not running subsequent tests, one could instead keep of how many tests have failed (if any) and print it in the end. If 0 failed test, all god, otherwise bad.